### DISCUSSION TITLE
Java and PlantUML: wrong extension `.latex`

### DISCUSSION THREAD

**Discussion by denisdamata** [plantuml/plantuml#1908] [General]
I am using Ubuntu 24 and trying to compile the following `temp.tex` file with the `plantuml` package (for testing purposes only):

```
\documentclass{article}
\usepackage{plantuml}

\begin{document}

Here is a UML diagram generated directly in LaTeX using Plant UML:

\begin{plantuml}
@startuml
Alice -> Bob: Test
Bobs --> Alice: Answer
@enduml
\end{plantuml}

\end{document}
```

To compile files with the `plantuml` package you need to use `lualatex` with the `-shell-escape` option:

```
dns@dns-pc:~/Downloads$ lualatex --synctex=1 --interaction=nonstopmode -shell-escape temp.tex
```
However, `java` uses the `-tlatex:nopreamble` option by default instead of the `-ttex:nopreamble` option which generates files with a `.latex` extension that `lualatex` does not recognize.

Here's an example of the error output:
```
...
Executing PlantUML... 
java -Djava.awt.headless=true -jar /usr/share/plantuml/plantuml.jar -charset UT
F-8 -tlatex:nopreamble temp-plantuml.txt

! LaTeX Error: File `temp-plantuml.tex' not found.

Type X to quit or <RETURN> to proceed,
or enter new name. (Default extension: tex)

Enter file name: 
! Emergency stop.
<read *> 
   
l.13 \end{plantuml}
                  
 1021 words of node memory still in use:
   4 hlist, 1 rule, 7 disc, 2 local_par, 2 dir, 14 glue, 6 kern, 1 penalty, 54 
glyph, 11 attribute, 48 glue_spec, 11 attribute_list, 2 temp, 2 write, 2 pdf_co
lorstack nodes
   avail lists: 2:6,3:1,5:1,9:1
!  ==> Fatal error occurred, no output PDF file produced!
Transcript written on temp.log.
```
There are several ways to work around this issue, but I want `java` to use the `-ttex:nopreamble` option by default and generate files with the `.tex` extension.

**Comment by Mk-arc** [comment]
Hi. I think i am now also experiencing this problem. Which is very strange. Since the project I am working with is old with its own (old) version of plantuml stored. 

I need to recreate the documents now  and suddenly this happens. No changes in plantuml version, no changes in the latex files. Before it worked.

What are your workarounds?

**Reply by Mk-arc** [reply]
![image](https://github.com/user-attachments/assets/054241a5-621f-404b-b739-1cde6b82726e)

Ok i did some digging through the latex plantuml plugin and figured that that the plantuml default extension name for the file sgot changed. The newer version from the latex plugin expects .tex ending, while my plantuml version was still old.

Solution: Use latest plantuml release.
